### PR TITLE
! Fixes position of ResultAction Menu

### PR DESCRIPTION
**Pull request by dguerinCoveo**
! Fixes alignment of icons inside ResultAction Menu
! Fixes left border bug on ResultAction Attach when Quick View is hidden
SNOW-284

### CODE REVIEW

**Comment by louis-bompart** [comment]
This looks CSS fixes specifics to ServiceNow, this should be done on ServiceNow repo I think.

File: pages/attached_result.html [hunk 1]
```diff
@@ -78,8 +78,8 @@
                 <script type="text/html" class="result-template">
                     <div class="coveo-result-frame">
                       <div class="CoveoResultActionsMenu">
-                        <div class="CoveoQuickview"></div>
                         <div class="CoveoAttachResult" data-attach-caption="Attach" data-detach-caption="Detach"></div>
+                        <div class="CoveoQuickview"></div>
```
Comment (louis-bompart) [inline pages/attached_result.html:82]: What's the change here?

File: src/sass/ResultActionMenu.scss [hunk 2]
```diff
@@ -32,7 +32,7 @@ $tooltip-z-index: 1000;
     cursor: pointer;
     display: none;
     position: relative;
-    height: $menu-height;
+    height: $menu-height+1px;
```
Comment (louis-bompart) [inline src/sass/ResultActionMenu.scss:35]: That seems a bit hackish.

File: src/sass/ResultActionMenu.scss [hunk 3]
```diff
@@ -41,6 +41,8 @@ $tooltip-z-index: 1000;
     margin-top: -$menu-height;
 
     &.coveo-menu-opened {
+        margin-top: 0px;
+        position: absolute;
```
Comment (louis-bompart) [inline src/sass/ResultActionMenu.scss:45]: Why the change in position?

File: src/sass/ResultActionMenu.scss [hunk 4]
```diff
@@ -51,7 +53,6 @@ $tooltip-z-index: 1000;
         height: $menu-height;
         position: relative;
         box-sizing: border-box;
-        border: thin solid $heather;
```
Comment (louis-bompart) [inline src/sass/ResultActionMenu.scss:54]: Why no border?

File: src/sass/ResultActionMenu.scss [hunk 5]
```diff
@@ -122,6 +123,11 @@ $tooltip-z-index: 1000;
     }
 }
 
+//Fix for attach icon aligment
+.CoveoAttachResult div.coveo-icon-attach{
```
Comment (louis-bompart) [inline src/sass/ResultActionMenu.scss:127]: Seems odd?

File: src/sass/ResultActionMenu.scss (hunk 2 above)
Comment (jeremierobert-coveo) [inline src/sass/ResultActionMenu.scss:35]: Where this 1px comes from ?

File: src/sass/ResultActionMenu.scss (hunk 4 above)
Comment (jeremierobert-coveo) [inline src/sass/ResultActionMenu.scss:54]: Why ?

File: src/sass/ResultActionMenu.scss [hunk 6]
```diff
@@ -41,6 +41,8 @@ $tooltip-z-index: 1000;
     margin-top: -$menu-height;
 
     &.coveo-menu-opened {
+        margin-top: 0px;
```
Comment (jeremierobert-coveo) [inline src/sass/ResultActionMenu.scss:44]: `margin-top: 0;` equivalent I think.

**Comment by dguerinCoveo** [comment]
Thanks for your comments. Here are the before/after Screenshots. 

Before:
Before_search_ui_extension_fixes

Before_search_ui_extension_fixes3


Before_search_ui_extension_fixes2


After:
After_search_ui_extension_fixes
After_search_ui_extension_fixes3
After_search_ui_extension_fixes2

**Comment by lvu285** [comment]
The fix is specific for ServiceNow, @jeremierobert-coveo @louis-bompart , can you double check it careful, as I'm afraid it might affect sfint :(

**Comment by mikegron** [comment]
The problem is there when using _only_ the extensions (locally), so it's not related to ServiceNow styling, although you are right that it could affect SF so we'll need to validate that 👯

File: pages/attached_result.html (hunk 1 above)
Comment (dguerinCoveo) [reply pages/attached_result.html:82]: There is a bug caused by the fact the the Quickview Icon is sometimes hidden in the ResultAction Menu. When the Quickview icon is hidden, its neighbor to the right still has a left border, creating a visual glitch. 

Placing the Quickview Icon at the far right of the  ResultAction Menu patches that bug, but does not fix it. 

I will revert that change so it does not hide the bug

File: src/sass/ResultActionMenu.scss (hunk 2 above)
Comment (dguerinCoveo) [reply src/sass/ResultActionMenu.scss:35]: Menu Menu Items' top are 1 pixel under the ResultAction Menu's border and both Menu Menu Items and Menu are the same height. Therefore, the Menu Menu Items left border ends 1 pixel under the ResultAction Menu.

I think that it will make more sense to make the Menu Menu Items 1 pixel shorter though, so I'll update that

File: src/sass/ResultActionMenu.scss (hunk 3 above)
Comment (dguerinCoveo) [reply src/sass/ResultActionMenu.scss:45]: My bad. I will revert that to the way it was

File: src/sass/ResultActionMenu.scss (hunk 4 above)
Comment (dguerinCoveo) [reply src/sass/ResultActionMenu.scss:54]: If you look closely at the before Screenshot, there are 2 different borders overlapping

File: src/sass/ResultActionMenu.scss (hunk 5 above)
Comment (dguerinCoveo) [reply src/sass/ResultActionMenu.scss:127]: I did not see the transparent border came from the %icon tag at line 15. I propose to remove the transparent border there since it is causing a 1 pixel shift with the Quickview Icon that does not have that transparent border

File: src/sass/ResultActionMenu.scss [hunk 7]
```diff
@@ -48,10 +48,9 @@ $tooltip-z-index: 1000;
     }
 
     .coveo-result-actions-menu-menu-item {
-        height: $menu-height;
+        height: $menu-height - 1px;
         position: relative;
```
Comment (mikegron) [inline src/sass/ResultActionMenu.scss:52]: @louis-bompart || @jeremierobert-coveo  C'est normal qu'ont override le style par défaut du jsui ici? Il me semble que le style par défaut du framework devrais marcher correctement. Par exemple si quelqu'un utilise le serach ui seul et décide d'ajouter les extensions, le styling vas changer.. Ont devrais juste modifier les trucs des extensions.

File: src/sass/ResultActionMenu.scss [hunk 8]
```diff
@@ -48,10 +48,9 @@ $tooltip-z-index: 1000;
     }
 
     .coveo-result-actions-menu-menu-item {
-        height: $menu-height;
+        height: $menu-height - 1px;
```
Comment (mikegron) [inline src/sass/ResultActionMenu.scss:51]: C'est sensé être -2 ici non? Vue que sa compte la border top et bottom.. Sinon, tu peux mettre box-sizing: content-box sur le menu, comme ca les borders serons a "l'extérieur" du menu, sa fait moins hacky je trouve..

Comment (dguerinCoveo) [reply src/sass/ResultActionMenu.scss:51]: Ca ferait moins hacky oui, sauf que si on met les borders du menu à l'extérieur, le menu fait soudainement 23 pixels de haut et il fait bouger le texte en dessous de lui parce qu'il empiète dessus et est en position: relative. 

Est-ce que la solution serait de mettre le menu en absolute?

File: src/sass/ResultActionMenu.scss (hunk 7 above)
Comment (louis-bompart) [reply src/sass/ResultActionMenu.scss:52]: Je suis d'accord que ce qui fait foi ca va être le standalone, donc ce que tu vois en dehors de SNOW ou SFINT. (Donc ce que tu vois dans gulp dev mostly)

Par contre si on suit ta logique @mikegron la PR devrait etre `rm src/sass/ResultActionMenu.scss` :P

**Comment by dguerinCoveo** [comment]
I pushed another commit this morning fixing the left border being present when there is only one ResultAction in the menu. This last commit reuses lines from salesforceintegration/src/develop/src/components/search-ui/ResultActionsMenu/scss/ResultActionsMenu.scss that fixes the same problem.

Here are the actual before/after screenshot taken from the search-extension-ui demo app and the salesforce app.

**Before on extension Demo App**

1
Before_search_ui_extension_fixes
2
Before_search_ui_extension_fixes3
3

Before_search_ui_extension_fixes2


**After on extension Demo App**

1
After_search_ui_extension_fixes
2
After_search_ui_extension_fixes3
3

After_search_ui_extension_fixes2



**Before on SF**
SF_before_1

**After on SF**
SF_after_1

It does not seem to impact SF since a lot of styling in that app are made inside the project.

I also think it is worth it to discuss the pertinence of having Css modifications of search-ui repo's components inside of the extension

**Comment by louis-bompart** [comment]
@jeremierobert-coveo Can you contextualize why we added this scss file in search-ui-extensions?

File: src/sass/ResultActionMenu.scss (hunk 8 above)
Comment (mikegron) [reply src/sass/ResultActionMenu.scss:51]: Oui je dirais de mettre absolute et valider avec SF.. C'est absolute par défaut dans le JS et meme sans les extra pixels, sa affecte le texte du result, ce qui ne devrais pas arriver!

Comment (mikegron) [reply src/sass/ResultActionMenu.scss:51]: Le menu devrais être 24px de haut incluant les border en fait

File: src/sass/ResultActionMenu.scss (hunk 7 above)
Comment (mikegron) [reply src/sass/ResultActionMenu.scss:52]: Pretty much :D ! Imo ont devrais juste garder les css qui sont spécifique aux components.. Il est trop tard?

File: src/sass/ResultActionMenu.scss [hunk 9]
```diff
@@ -4,6 +4,7 @@ $menu-height: 22px;
 $menu-radius: 5px;
 $tooltip-arrow-length: 6px;
 $tooltip-z-index: 1000;
+$result-action-menu-item-border: thin solid #bcc3ca;
```
Comment (jeremierobert-coveo) [inline src/sass/ResultActionMenu.scss:7]: variable for #bcc3ca 😛

File: src/sass/ResultActionMenu.scss (hunk 7 above)
Comment (louis-bompart) [reply src/sass/ResultActionMenu.scss:52]: > Pretty much :D ! Imo ont devrais juste garder les css qui sont spécifique aux components.. Il est trop tard?

Well, @jeremierobert-coveo pourraît fournir le contexte, mais en vérité le JSUI devrait pas own le ResultActionMenu genre

**Comment by dguerinCoveo** [comment]
@louis-bompart @jeremierobert-coveo I just want to bring this back to your attention. Do you guys think I should make this PR here, in the search-ui repo or in the Service Now repo?

**Comment by louis-bompart** [comment]
@dguerinCoveo, following up on our meeting moments ago, I let you the honor of closing down this PR :pressf:
